lib/inc/hocon/config_value.hpp

```diff
@@ -163,8 +169,32 @@ namespace hocon {
         private:
             std::string _prefix;
         };
+        void require_not_ignoring_fallbacks() const;
+
+        /* this is virtualized rather than a field because only some subclasses
+         * really need to store the boolean, and they may be able to pack it
+         * with another boolean to save space.
+         */
+        bool ignores_fallbacks() const;
+        shared_value with_fallbacks_ignored() const;
+
+        shared_value merged_with_the_unmergeable(std::vector<shared_value> const& stack,
+                                                 std::shared_ptr<const unmergeable> fallback) const;
+        shared_value merged_with_the_unmergeable(std::shared_ptr<const unmergeable> fallback) const;
+
+        shared_value merged_with_object(std::vector<shared_value> const& stack, shared_object fallback) const;
+        shared_value merged_with_object(shared_object fallback) const;
+
+        shared_value merged_with_non_object(std::vector<shared_value> const& stack, shared_value fallback) const;
+        shared_value merged_with_non_object(shared_value fallback) const;
+
+        shared_value construct_delayed_merge(shared_origin origin, std::vector<shared_value> stack) const;
```

Comment:
The only reason this function exists is to be overridden by config_object. So it should be virtual.